.github/workflows/kind.yml

```diff
@@ -64,13 +64,30 @@ jobs:
         export GOPATH=$(go env GOPATH)
         make wait-for-work-agent
         make e2e-test
-    
+
+    - name: Clean up clusters
+      run: |
+        RUN_MODE="delete" ./build/manage-clusters.sh
+
+    - name: Create KinD Cluster (${{ matrix.kind }}) and deploy components in hosted mode
+      env:
+        KIND_VERSION: ${{ matrix.kind }}
+        HOSTED_MODE: "true"
```

Comment:
Optional: We should add this to the matrix so that these tests can run in parallel.

Reply:
Good idea! I just changed the GitHub actions for that.